### PR TITLE
bpo-44077: Expose IP_RECVTOS in the socket module

### PR DESCRIPTION
**Pull request by gsauthof**
https://bugs.python.org/issue44077

https://bugs.python.org/issue44077

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@gsauthof

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by corona10 — CHANGES_REQUESTED** [review]
Please update Docs and NEWS

For your help: https://github.com/python/cpython/pull/24946/files and https://docs.python.org/3/library/socket.html?highlight=socket#socket.SOMAXCONN

**Comment by gsauthof** [comment]
I have made the requested changes; please review again.

**Comment by terryjreedy** [comment]
@pablogsal Pablo, unless you are willing for this new constant to be backported to 3.10.0b2 (or b3 if it misses b2), the What's new entry should go in the currently non-existent What's new in 3.11.  Do you plan to add it soon?  I am not going to ask a new contributor to do so.

Georg, the 'new in' entry may need to be changed depending on Pablo's answer.

@corona10 Is this something you might merge, or do we need to find a socket person (I am not)?

**Comment by corona10** [comment]
@terryjreedy 
I think that if this PR is merged, this changed only affects 3.11

**Comment by corona10** [comment]
> Is this something you might merge, or do we need to find a socket person

I often observe that this kind of constants adding was accepted.
But getting opinions from socket experts looks great.

File: Doc/library/socket.rst [hunk 1]
```diff
@@ -379,6 +379,9 @@ Constants
       On Windows, ``TCP_KEEPIDLE``, ``TCP_KEEPINTVL`` appear if run-time Windows
       supports.
 
+   .. versionchanged:: 3.10
```
Comment (corona10) [inline Doc/library/socket.rst:382]:
```suggestion
   .. versionchanged:: 3.11
```

**Review by corona10 — CHANGES_REQUESTED** [review]

**Comment by corona10** [comment]
@terryjreedy 
FYI, I did not decide to merge this PR or not. Before accepting this PR we should investigate the use cases of `IP_RECVTOS`.
And if the socket expert exists, we can get a review from the expert.

**Comment by gsauthof** [comment]
I have made the requested changes; please review again.

**Comment by pablogsal** [comment]
> @pablogsal Pablo, are you willing for this new constant to be backported to 3.10.0b2 ?

Sorry for the late response, This has flown under my radar :(

I am ok including this constant, as it is contained enough

**Pull request merged**